src/scheduler.cpp

```diff
@@ -195,7 +195,11 @@ void SingleThreadedSchedulerClient::AddToProcessQueue(std::function<void ()> fun
         LOCK(m_cs_callbacks_pending);
         m_callbacks_pending.emplace_back(std::move(func));
     }
-    MaybeScheduleProcessQueue();
+    if (m_pscheduler->AreThreadsServicingQueue()) {
```

Comment:
In commit "Handle no threads serving in MaybeScheduleProcessQueue" (03cd591f47f852e0a2c2e6262367eff4acefca62)

Was experimenting and wrote a small test for this change:

```c++
// Ensure scheduled tasks run even when there are no scheduler threads
BOOST_AUTO_TEST_CASE(singlethreadedscheduler_nothreads)
{
    CScheduler scheduler;
    SingleThreadedSchedulerClient client(&scheduler);
    bool task_ran = false;
    client.AddToProcessQueue([&] { task_ran = true; });
    BOOST_CHECK(task_ran);
}
```

Could add this to scheduler_tests.cpp. It fails without this commit and passes with it.